app/models/concerns/github_repoable.rb

```diff
@@ -8,14 +8,18 @@ def add_team_to_github_repository
     github_repository = GitHubRepository.new(organization.github_client, github_repo_id)
     github_team       = GitHubTeam.new(organization.github_client, github_team_id)
 
-    github_team.add_team_repository(github_repository.full_name)
+    options = {}
+    options[:permission] = 'admin' if give_admin_permission?
+    github_team.add_team_repository(github_repository.full_name, options)
   end
 
   def add_user_as_collaborator
     github_user = GitHubUser.new(user.github_client, user.uid)
     repository  = GitHubRepository.new(organization.github_client, github_repo_id)
 
-    delete_github_repository_on_failure { repository.add_collaborator(github_user.login) }
+    options = {}
+    options[:permission] = 'admin' if give_admin_permission?
+    delete_github_repository_on_failure { repository.add_collaborator(github_user.login, options) }
   end
 
   # Public
```

Comment:
So originally I had started to write it where `GitHubRepoable` referenced `assignment` or `group_assignment`, but I noticed that no other methods in the file reference either of those. That made me think that it'd be better to keep it that way, keeping `GitHubRepoable` more generic and reusable. This means that if we ever added something other than an `Assignment` or `GroupAssignment` that needed to create a repository, we'd just have to implement `give_admin_permission?` on that thing without needing to change `GitHubRepoable` at all. Similarly, we could conceivably add a "make all my assignments default to making students be repo admins unless otherwise specified" setting to organizations. Would it make sense for `GitHubRepoable` to include that logic too, or should it only ask/care about "should I set the admin permission or not"? Of course, we may never make these kinds of changes, but it still seemed nice to decrease the coupling that way.

Anyway, just wanted to explain my line of thinking there. If you don't think those concerns matter, I can move `give_admin_permission?` into `GitHubRepoable`.